From 33ba9ca4b789b49c4162c1e6449e4b0449d35b49 Mon Sep 17 00:00:00 2001
From: Al Viro <viro@zeniv.linux.org.uk>
Date: Tue, 11 Jun 2013 12:52:02 +0400
Subject: [PATCH 097/132] ext3 ->tmpfile() support

In this case we do need a bit more than usual, due to orphan
list handling.

Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Change-Id: I037ce93117ab39de6450fc6593b36511852b05f7

# Conflicts:
#	fs/ext3/namei.c
---
 fs/ext3/namei.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/ext3/namei.c b/fs/ext3/namei.c
index 4081189e089..74c17351fe5 100644
--- a/fs/ext3/namei.c
+++ b/fs/ext3/namei.c
@@ -1777,11 +1777,11 @@ retry:
 		inode->i_op = &ext3_file_inode_operations;
 		inode->i_fop = &ext3_file_operations;
 		ext3_set_aops(inode);
-		d_tmpfile(dentry, inode);
 		err = ext3_orphan_add(handle, inode);
 		if (err)
 			goto err_drop_inode;
 		mark_inode_dirty(inode);
+		d_tmpfile(dentry, inode);
 		unlock_new_inode(inode);
 	}
 	ext3_journal_stop(handle);
-- 
2.17.1

